src/freertos_drivers/st/Stm32Can.cxx

```diff
@@ -44,72 +44,141 @@
 #if defined (STM32F072xB) || defined (STM32F091xC)
 
 #include "stm32f0xx_hal_cortex.h"
-#define CAN_IRQN CEC_CAN_IRQn
-
+#define CAN1_IRQN CEC_CAN_IRQn
+#define CAN1 CAN
 #define CAN_CLOCK cpu_clock_hz
 
 #elif defined (STM32F103xB)
 
 #include "stm32f1xx_hal_cortex.h"
 #define SPLIT_INT
-#define CAN_TX_IRQN USB_HP_CAN1_TX_IRQn
-#define CAN_IRQN CAN_TX_IRQN
-#define CAN_SECOND_IRQN USB_LP_CAN1_RX0_IRQn
-#define CAN_THIRD_IRQN CAN1_SCE_IRQn
-#define CAN CAN1
+#define CAN1_TX_IRQN USB_HP_CAN1_TX_IRQn
+#define CAN1_IRQN CAN1_TX_IRQN
+#define CAN1_SECOND_IRQN USB_LP_CAN1_RX0_IRQn
+#define CAN1_THIRD_IRQN CAN1_SCE_IRQn
+
+#ifdef CAN2
+#define CAN2_TX_IRQN USB_HP_CAN2_TX_IRQn
```

Comment:
The correct values here are
`CAN2_TX_IRQn`
`CAN2_RX0_IRQn`
`CAN2_SCE_IRQn`

(correct value comes from STM32CubeFW_F1/Drivers/CMSIS/Devices/Include/stm32f105xc.h
Technically an F103 never has two CAN controllers. For F105 and F107 the CAN1 declarations are not correct, since they have separate USB and CAN interrupts, but it's ok to put ifdef can2 here.)

Reply:
Should I have a separate elifs for the  F105 and F107s?  There does not appear to be defines for these, just for the F103...

Reply:
OK, fixed, pending whether to add elifs for the F105 and F107.